src/Controller/AppController.php

```diff
@@ -55,18 +55,21 @@ public function initialize()
     /**
      * Before render callback.
      *
-     * @param \Cake\Event\Event $event The beforeRender event.
+     * @param  \Cake\Event\Event $event The beforeRender event.
      * @return \Cake\Http\Response|null|void
      */
     public function beforeRender(Event $event)
     {
         // Note: These defaults are just to get started quickly with development
-        // and should not be used in production. You should instead set "_serialize"
-        // in each action as required.
-        if (!array_key_exists('_serialize', $this->viewVars) &&
-            in_array($this->response->type(), ['application/json', 'application/xml'])
-        ) {
-            $this->set('_serialize', true);
+        // and should not be used in production.
+        // You should instead set "_serialize" in each action as required.
+        if (!array_key_exists('_serialize', $this->viewVars)) {
+            $contentType = $this->response->getHeaderLine('Content-Type');
+            if (strstr($contentType, 'application/json') ||
+                strstr($contentType, 'application/xml')
```

Comment:
These `strstr()` calls could be `strpos()`. It is a bit faster and uses less memory. While it is likely not relevant here it is a good practice to adopt.

Reply:
Yeah it's a beautiful confusion, I had planned to user `strpos()` 😢
May I force push ?

Reply:
Of course. Force push and rebase as much as you need.